apis/apps/v1alpha1/componentdefinition_types.go

```diff
@@ -472,25 +472,36 @@ type LifecycleActionHandler struct {
 
 	// customHandler defines the custom way to do action.
 	// +optional
-	CustomHandler *Action `json:"customHandler,omitempty"`
+	CustomHandler *CustomHandler `json:"customHandler,omitempty"`
+}
+
+type CustomHandler struct {
+	// action defines the custom action to take.
+	// +kubebuilder:validation:Required
+	Action *Action `json:"action,omitempty"`
+
+	// scriptSpecSelectors defines the selector of the scriptSpecs that need to be referenced.
+	// Once ScriptSpecSelectors is defined, the scripts defined in scripts can be referenced in the CustomHandler Action.
+	// +optional
+	ScriptSpecSelectors []ScriptSpecSelector `json:"scriptSpecSelectors,omitempty"`
```

Comment:
> what's the purpose of this field?

The purpose of this field is originally intended for reusing scripts defined in scriptSpec and simplifying the script writing in actions (so that they can be managed centrally in scriptSpec). Therefore, the field was directly reused from the previous version's definition. 

Perhaps it can be optimized to use a similar approach  like Vars references in componentDefinition, where the scripts in scriptSpec can be referenced, or a set of rules specifically designed for managing scripts or commands in lifecycle action scenario.

what's your suggestion?

Reply:
If you want to use commands defined in scripts, you can just setup the Action with TargetPodSelector, MatchingKey and Container set appropriately.